Src/Plugins/Security/SEC101_001.HttpAuthorizationRequestHeaderValidator.cs

```diff
@@ -42,25 +43,26 @@ static HttpAuthorizationRequestHeaderValidator()
         }
 
         protected override IEnumerable<ValidationResult> IsValidStaticHelper(ref string matchedPattern,
-                                                                             Dictionary<string, string> groups)
+                                                                             Dictionary<string, FlexMatch> groups)
         {
-            if (!groups.TryGetNonEmptyValue("host", out string host) ||
-                !groups.TryGetNonEmptyValue("scheme", out string scheme) ||
-                !groups.TryGetNonEmptyValue("secret", out string secret))
+            if (!groups.TryGetNonEmptyValue("host", out FlexMatch host) ||
+                !groups.TryGetNonEmptyValue("scheme", out FlexMatch scheme) ||
+                !groups.TryGetNonEmptyValue("secret", out FlexMatch secret))
             {
                 return ValidationResult.CreateNoMatch();
             }
 
-            groups.TryGetNonEmptyValue("path", out string path);
+            groups.TryGetNonEmptyValue("path", out FlexMatch path);
 
             var validationResult = new ValidationResult
             {
+                RegionFlexMatch = secret,
                 Fingerprint = new Fingerprint
                 {
-                    Host = host,
-                    Path = path,
-                    Scheme = scheme,
-                    Secret = secret,
+                    Host = host.Value,
+                    Path = path?.Value,
```

Comment:
>path?.Value [](http://example.com/codeflow?start=27&length=11)

Here's an important outcome of this change, if a value, such as 'path' in this case, is optional, the flex match could itself be null. It's too bad that we've introduced this consideration into the design. We could try to do something clever in an implicit cast operator to try to overcome it, though, not strictly sure on feasibility without looking at it.